File: Lab1/Lab1/Lab1/model/CPU.cs

using Lab1.dto;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab1Code.model
{
    public class CPU 
    {
        private CPUDto cpuDtoAdaptee;

        public CPU(CPUTypes cpuTypes, int usingPover)
        {
            this.cpuDtoAdaptee = new CPUDto(cpuTypes, usingPover);
        }

        public CPU(CPUDto cPuDto)
        {
            this.cpuDtoAdaptee = cPuDto;
        }

        public CPUDto CpuDtoAdaptee
        {
            get => cpuDtoAdaptee;
            set => cpuDtoAdaptee = value;
        }

        public CPUTypes CpuTypes
        {
            get => cpuDtoAdaptee.CpuTypes; 
            set => cpuDtoAdaptee.CpuTypes = value;
        }

        public int UsingPover
        {
            get => cpuDtoAdaptee.UsingPover;
            set => cpuDtoAdaptee.UsingPover = value;
        }
    }
}

File: Lab1/Lab1/lab1j/src/model/CPU.java
package model;

import dto.CPUDto;
import dto.CPUTypes;

public class CPU implements IAdapter<CPUDto>
    {
        private CPUDto cpuDtoAdaptee;

        public CPU(CPUTypes cpuTypes, int usingPover)
        {
            this.cpuDtoAdaptee = new CPUDto(cpuTypes, usingPover);
        }

        public CPU(CPUDto cPuDto)
        {
            this.cpuDtoAdaptee = cPuDto;
        }

        public CPUDto CpuDtoAdaptee()
        {
            return null;
        }

        public CPUTypes CpuTypes()
        {return null;
        }

        public int UsingPover()
        {
            return 1;
        }

        public CPUDto GetAdaptee()
        {
            return cpuDtoAdaptee;
        }

        public void SetAdaptee(CPUDto adaptee)
        {
            cpuDtoAdaptee=adaptee;
        }

        public CPU()
        {
        }
    }

